dwi_ml/data/processing/streamlines/data_augmentation.py

```diff
@@ -1,75 +1,62 @@
+import logging
 from typing import List, Union
 
 import nibabel as nib
 import numpy as np
 
+from dipy.io.stateful_tractogram import StatefulTractogram
 from scipy.stats import truncnorm
-from dwi_ml.data.processing.space.convert_world_to_vox import (
-    convert_world_to_vox)
+
 from dwi_ml.data.processing.streamlines.utils import split_array_at_lengths
 
 
-def add_noise_to_streamlines(
-        streamlines: Union[nib.streamlines.ArraySequence, np.ndarray],
-        noise_sigma: float, noise_rng: np.random.RandomState,
-        convert_mm_to_vox: bool = False, affine: np.ndarray = None)\
-        -> Union[nib.streamlines.ArraySequence, np.ndarray]:
+# Checked!
+def add_noise_to_streamlines(sft: StatefulTractogram,
+                             noise_sigma: float,
+                             noise_rng: np.random.RandomState):
     """Add gaussian noise (truncated to +/- 2*std) to streamlines coordinates
-     *in-place*
 
     Parameters
     ----------
-    streamlines : nib.streamlines.ArraySequence
+    sft : StatefulTractogram
         Streamlines.
     noise_sigma : float
         Standard deviation of the gaussian noise to add to the streamlines.
-        NOTE: This value should be in the same space as the streamlines
-        (assuming the space is isometric)
+        CAREFUL. With do not deal with space. Make sure your noise is in the
+        same space as your sft.
     noise_rng : np.random.RandomState object
         Random number generator.
-    convert_mm_to_vox: bool
-        If true, will convert noise from mm to vox_iso using affine first.
-        Note that we don't support vox to mm yet, nor mm to vox_noniso.
-        [False]
-    affine: np.ndarray
-        Needed if convert_noise_space is True. Ex : affine_vox2rasmm
 
     Returns
     -------
-    streamlines : nib.streamlines.ArraySequence
-        Noisy streamlines.
+    noisy_sft : StatefulTractogram
+        Noisy streamlines. Note. Adding noise may create invalid streamlines
+        (i.e. out of the box in voxel space). If you want to save noisy_sft,
+        please perform noisy_sft.remove_invalid_streamlines() first.
     """
+    logging.info("Please note your sft space is in {}. We suppose that the "
+                 "noise, {}, fits.".format(sft.space, noise_sigma))
+    
+    # Perform noise addition (flattening before to go faster)
+    flattened_coords = np.concatenate(sft.streamlines, axis=0)
+    flattened_coords += truncnorm.rvs(-2, 2, size=flattened_coords.shape,
```

Comment:
I used a truncated normal distribution, clipped between -2 and 2... Is this still desirable? We could clip to `+- 2*noise` for example

Reply:
Not sure the rationale behind using a truncated a normal distribution vs. one that has not been truncated.

May I assume that the rationale behind using a truncated distribution was due to the fact that it was been observed/we assume that beyond those limits there is no "noticeable" effect of the noise (i.e. to save computations that hardly perform any "noticeable" data augmentation)?

And since this is a normal distribution (sigma is 1), the +/2 limits mean that the noise is truncated to +/- 2*sigma. This seems reasonable, since roughly 95% of the CDF for a normal distribution is within +/- 2*sigma.

However, I am not sure about how the range impacts the output (i.e. I ignore what the range of the input values is).

Also, according to the documentation this adds "Gaussian" noise. Thus, we should add the mean and variance/std dev as parameters (somehow checking that the values or ranges make sense?)

I am surely missing the whole picture.